pkg/logs/config/source.go

```diff
@@ -57,3 +61,10 @@ func (s *LogSource) GetInputs() []string {
 	}
 	return inputs
 }
+
+// SetParser adds a parser
+func (s *LogSource) SetParser(parser parser.Parser) {
+	s.lock.Lock()
+	s.Parser = parser
+	s.lock.Unlock()
+}
```

Comment:
As explained IRL, it could be worth having a thread-safe `GetParser()`, but currently it would not change anything.